bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/widgets/Control.java

```diff
@@ -4885,20 +4885,18 @@ LRESULT WM_DESTROY (long wParam, long lParam) {
 LRESULT WM_DPICHANGED (long wParam, long lParam) {
 	// Map DPI to Zoom and compare
 	int nativeZoom = DPIUtil.mapDPIToZoom (OS.HIWORD (wParam));
-	int newSWTZoom = DPIUtil.getZoomForAutoscaleProperty (nativeZoom);
-	int oldSWTZoom = getShell().getZoom();
+	int oldNativeZoom = getShell().getNativeZoom();
 
 	// Throw the DPI change event if zoom value changes
-	if (newSWTZoom != oldSWTZoom) {
+	if (nativeZoom != oldNativeZoom) {
 		Event event = new Event();
 		event.type = SWT.ZoomChanged;
 		event.widget = this;
-		event.detail = newSWTZoom;
+		event.detail = nativeZoom;
 		event.doit = true;
 
 		if (DPIUtil.isAutoScaleOnRuntimeActive()) {
 			DPIUtil.setDeviceZoom (nativeZoom);
-			getShell().setNativeZoom(nativeZoom);
```

Comment:
Just out of curiosity: why was it sufficient to set the native but not the current zoom here and why is it okay to completely remove it?

Reply:
Thanks for the explanation. That makes the changes of this PR even more valuable in terms of reducing responsibilities for propagating zoom changes.